From f580090ea22f2d214e0b85aeda8572342fc9495d Mon Sep 17 00:00:00 2001
From: Jung Zhao <jung.zhao@rock-chips.com>
Date: Tue, 6 Feb 2018 14:28:56 +0800
Subject: [PATCH] video: rockchip: vpu: fix crash bug when translate failed

when translate failed, driver will print the regs.
but kernel must use kernel memory, if we print user
memory, it will crash.

Change-Id: Icfe30e4b70485e2d20d0d75dca7d51d3b904e7cc
Signed-off-by: Jung Zhao <jung.zhao@rock-chips.com>
---
 drivers/video/rockchip/vcodec/vcodec_service.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/vcodec/vcodec_service.c b/drivers/video/rockchip/vcodec/vcodec_service.c
index 4b2eb9093d35..f30777925b03 100644
--- a/drivers/video/rockchip/vcodec/vcodec_service.c
+++ b/drivers/video/rockchip/vcodec/vcodec_service.c
@@ -1235,7 +1235,7 @@ static struct vpu_reg *reg_init(struct vpu_subdev_data *data,
 		vpu_err("error: translate reg address failed, dumping regs\n");
 		for (i = 0; i < size >> 2; i++)
 			dev_err(pservice->dev, "reg[%02d]: %08x\n",
-				i, *((u32 *)src + i));
+				i, reg->reg[i]);
 
 		kfree(reg);
 		return NULL;
-- 
2.35.3

